File: Handler/Command/TeleportCommand.java
package com.countrygamer.countrygamer_core.Handler.Command;

import java.util.ArrayList;
import java.util.List;

import net.minecraft.command.ICommand;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.server.MinecraftServer;

import com.countrygamer.countrygamer_core.Core;
import com.countrygamer.countrygamer_core.lib.CoreUtil;

public class TeleportCommand implements ICommand {

	private List aliases;

	public TeleportCommand() {
		this.aliases = new ArrayList();
		this.aliases.add("cgctp");
	}

	@Override
	public String getCommandName() {
		return "Core Teleport";
	}

	@Override
	public String getCommandUsage(ICommandSender icommandsender) {
		return "cgctp <playername> <dimension name> <x> <y> <z>";
	}

	@Override
	public List getCommandAliases() {
		return this.aliases;
	}

	@Override
	public void processCommand(ICommandSender icommandsender, String[] astring) {
		if (astring.length == 0) {
			// TODO
			// icommandsender.sendChatToPlayer((new ChatMessageComponent())
			// .addText("Invalid Arguments"));
			return;
		}
		String playerName = astring[0];
		String dimensionName = astring[1];
		double x, y, z;
		try {
			x = Double.parseDouble(astring[2]);
			y = Double.parseDouble(astring[3]);
			z = Double.parseDouble(astring[4]);
		} catch (NumberFormatException e) {
			// TODO
			// icommandsender.sendChatToPlayer((new ChatMessageComponent())
			// .addText("Invalid Arguments"));
			if (Core.DEBUG) {
				Core.log.info("Cannot parse arguments");
				System.err.println(e.getStackTrace());
			}
			return;
		}
		EntityPlayerMP playerMP = MinecraftServer.getServer()
				.getConfigurationManager().getPlayerForUsername(playerName);
		// CG_Core.instance.dimLoad();
		if (playerMP != null && Core.dimensions.get(dimensionName) != null) {
			CoreUtil.teleportPlayerToDimension(playerMP,
					Core.dimensions.get(dimensionName));
			CoreUtil.teleportPlayer(playerMP, x, y, z, false, false);
		} else {
			// TODO
			// icommandsender.sendChatToPlayer((new ChatMessageComponent())
			// .addText("Player " + playerName + " is not availible."));
		}
	}

	@Override
	public boolean canCommandSenderUseCommand(ICommandSender icommandsender) {
		return true;
	}

	@Override
	public List addTabCompletionOptions(ICommandSender icommandsender,
			String[] astring) {
		return null;
	}

	@Override
	public boolean isUsernameIndex(String[] astring, int i) {
		return false;
	}

	@Override
	public int compareTo(Object o) {
		return 0;
	}

}

File: src/main/scala/com/temportalist/origin/library/common/command/TeleportCommand.scala
package com.temportalist.origin.library.common.command

import java.util

import com.temportalist.origin.library.common.Origin
import com.temportalist.origin.library.common.utility.{Player, Teleport}
import net.minecraft.command.{CommandBase, ICommandSender}
import net.minecraft.entity.player.EntityPlayerMP

/**
 *
 *
 * @author CountryGamer
 */
object TeleportCommand extends CommandBase {

	var aliases: util.List[String] = new util.ArrayList[String]()
	this.aliases.add("origin")

	override def getCommandName: String = {
		this.aliases.get(0)
	}

	override def getCommandAliases: util.List[_] = {
		this.aliases
	}

	override def getCommandUsage(p1: ICommandSender): String = {
		"origin [tp [x] [y] [z] <Dimension Name>:null]"
	}

	override def processCommand(sender: ICommandSender, args: Array[String]): Unit = {
		/*
		if (args.length == 0) {
			return
		}

		val playerName: String = args(0)
		val dimensionName: String = args(1)
		var x: Double = 0.0
		var y: Double = 0.0
		var z: Double = 0.0
		try {
			x = args(2).toDouble
			y = args(3).toDouble
			z = args(4).toDouble
		}
		catch {
			case e: NumberFormatException => {
				if (commandSender.isInstanceOf[EntityPlayer])
					Player.sendMessageToPlayer(commandSender.asInstanceOf[EntityPlayer],
						"Cannot parse arguments")
			}
		}
		*/
		if (args.length == 0) {
			return
		}

		val commandType: String = args(0)

		if (commandType.equals("tp")) {
			// origin tp ...
			sender match {
				case player1: EntityPlayerMP =>
					var player: EntityPlayerMP = player1

					if (args.length == 1) {
						Teleport.toCursorPosition(player, 20000.0D)
						return
					}

					// origin tp x      y z
					// origin tp x      y z dimName
					// origin tp player x y z
					// origin tp player x y z       dimName

					if (args.length < 4) {
						Player.message(player, "Not enough arguments")
						return
					}

					var hasAlternatePlayer: Boolean = false
					try {
						val potentialXCoord: Double = args(1).toDouble
					}
					catch {
						case e: NumberFormatException =>
							hasAlternatePlayer = true
					}

					var coordStartIndex: Int = 1
					if (hasAlternatePlayer) {
						coordStartIndex += 1
						player = Player.getPlayer(args(1))
					}

					var x: Double = 0.0D
					var y: Double = 0.0D
					var z: Double = 0.0D
					try {
						x = args(coordStartIndex + 0).toDouble
						y = args(coordStartIndex + 1).toDouble
						z = args(coordStartIndex + 2).toDouble
					}
					catch {
						case e: NumberFormatException =>
							Player.message(player1,
								"Something went terribly wrong...")
							return
					}

					if ((!hasAlternatePlayer && args.length > 4) ||
							(hasAlternatePlayer && args.length > 5)) {
						val dimName: String = args(coordStartIndex + 3)

						if (!Origin.dimensions.containsKey(dimName)) {
							Player.message(player1,
								"That was not a valid dimension name!")
							return
						}

						val dimID: Int = Origin.dimensions.get(dimName)

						Teleport.toDimension(player, dimID)

					}

					Teleport.toPoint(player, x, y, z)

				case _ =>
			}
		}

	}

}
